usecases/objects/batch_delete.go

```diff
@@ -46,6 +46,28 @@ func (b *BatchManager) DeleteObjects(ctx context.Context, principal *models.Prin
 	return b.deleteObjects(ctx, principal, match, dryRun, output, repl, tenant)
 }
 
+// DeleteObjectsFromGRPC deletes objects in batch based on the match filter
+func (b *BatchManager) DeleteObjectsFromGRPC(ctx context.Context, principal *models.Principal,
+	params BatchDeleteParams,
+	repl *additional.ReplicationProperties, tenant string,
+) (BatchDeleteResult, error) {
+	err := b.authorizer.Authorize(principal, "delete", "batch/objects")
+	if err != nil {
+		return BatchDeleteResult{}, err
+	}
+
+	unlock, err := b.locks.LockConnector()
+	if err != nil {
+		return BatchDeleteResult{}, NewErrInternal("could not acquire lock: %v", err)
+	}
+	defer unlock()
```

Comment:
shouldn't we run `unlock()` method also in case of error?

Reply:
I copied that from the function above `DeleteObjects` - I assume we didn't get the lock if there was an error.

I quickly checked other usage of `m.locks.LockConnector()`and they all do the same